src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/categories/CategorySyncIT.java

```diff
@@ -0,0 +1,496 @@
+            .build();
+        categorySync = new CategorySync(categorySyncOptions);
+    }
+
+    /**
+     * Cleans up the target and source test data that were built in this test class.
+     */
+    @AfterClass
+    public static void tearDown() {
+        deleteAllCategories(CTP_TARGET_CLIENT);
+        deleteAllCategories(CTP_SOURCE_CLIENT);
+        deleteTypesFromTargetAndSource();
+    }
+
+    @Test
+    public void syncDrafts_withChangesOnly_ShouldUpdateCategories() {
+        createCategories(CTP_SOURCE_CLIENT, getCategoryDraftsWithPrefix(Locale.ENGLISH, "new",
+            null, 2));
+
+        final List<Category> categories = CTP_SOURCE_CLIENT
+            .execute(CategoryQuery.of()
+                                  .withLimit(SphereClientUtils.QUERY_MAX_LIMIT)
+                                  .withExpansionPaths(ExpansionPath.of("custom.type"))
+                                  .plusExpansionPaths(CategoryExpansionModel::parent)
+            )
+            .toCompletableFuture().join().getResults();
+
+        // Put the keys in the reference ids to prepare for reference resolution
+        final List<CategoryDraft> categoryDrafts = replaceCategoriesReferenceIdsWithKeys(categories);
+
+        final CategorySyncStatistics syncStatistics = categorySync.sync(categoryDrafts).toCompletableFuture().join();
+
+        assertThat(syncStatistics.getReportMessage())
+            .isEqualTo(format("Summary: %d categories were processed in total (%d created, %d updated, %d failed to "
+                + "sync and %s categories with a missing parent).", 2, 0, 2, 0, 0));
+
+        assertThat(callBackErrorResponses).isEmpty();
+        assertThat(callBackExceptions).isEmpty();
+        assertThat(callBackWarningResponses).isEmpty();
+    }
+
+    @Test
+    public void syncDrafts_withNewCategories_ShouldCreateCategories() {
+        createCategories(CTP_SOURCE_CLIENT, getCategoryDraftsWithPrefix(Locale.ENGLISH, "new",
+            null, 3));
+
+        final List<Category> categories = CTP_SOURCE_CLIENT
+            .execute(CategoryQuery.of()
+                                  .withLimit(SphereClientUtils.QUERY_MAX_LIMIT)
+                                  .withExpansionPaths(ExpansionPath.of("custom.type"))
+                                  .plusExpansionPaths(CategoryExpansionModel::parent)
+            )
+            .toCompletableFuture().join().getResults();
+
+        // Put the keys in the reference ids to prepare for reference resolution
+        final List<CategoryDraft> categoryDrafts = replaceCategoriesReferenceIdsWithKeys(categories);
+
+        final CategorySyncStatistics syncStatistics = categorySync.sync(categoryDrafts).toCompletableFuture().join();
+
+        assertThat(syncStatistics.getReportMessage())
+            .isEqualTo(format("Summary: %d categories were processed in total (%d created, %d updated, %d failed to "
+                + "sync and %s categories with a missing parent).", 3, 1, 2, 0, 0));
```

Comment:
The test name says that new categories should be created but it asserts that 1 category is created, another 2 are updated. Could you help me to understand why 2 categories were updated?

Reply:
It's because this test method creates a 3 new categories in the source project  with name and description ("newName" and "newDescription") and the keys are always being incremented as: "key1", "key2", "key3", etc.. So the first two categories with ("key1" and "key2") will be the same as the ones in the target project but with new names and descriptions and thus the 2 updates. The target project doesn't have a third category at all. Therefore the category with key "key3" will be created :)